packages/run-protocol/src/vaultFactory/vault.js

```diff
@@ -329,35 +444,47 @@ export const makeVaultKit = (
       );
     } else if (proposal.give.RUN) {
       // We don't allow runDebt to be negative, so we'll refund overpayments
-      const acceptedRun = AmountMath.isGTE(proposal.give.RUN, runDebt)
-        ? runDebt
+      const acceptedRun = AmountMath.isGTE(proposal.give.RUN, getDebtAmount())
+        ? getDebtAmount()
         : proposal.give.RUN;
 
       vaultSeat.incrementBy(seat.decrementBy(harden({ RUN: acceptedRun })));
     }
   };
 
-  // Calculate the fee, the amount to mint and the resulting debt.
+  /**
+   * Calculate the fee, the amount to mint and the resulting debt
+   *
+   * @param {ProposalRecord} proposal
+   * @param {{vault: Amount, client: Amount}} runAfter
+   */
   const loanFee = (proposal, runAfter) => {
     let newDebt;
+    const currentDebt = getDebtAmount();
     let toMint = AmountMath.makeEmpty(runBrand);
     let fee = AmountMath.makeEmpty(runBrand);
     if (proposal.want.RUN) {
       fee = ceilMultiplyBy(proposal.want.RUN, manager.getLoanFee());
       toMint = AmountMath.add(proposal.want.RUN, fee);
-      newDebt = AmountMath.add(runDebt, toMint);
+      newDebt = AmountMath.add(currentDebt, toMint);
     } else if (proposal.give.RUN) {
-      newDebt = AmountMath.subtract(runDebt, runAfter.vault);
+      newDebt = AmountMath.subtract(currentDebt, runAfter.vault);
     } else {
-      newDebt = runDebt;
+      newDebt = currentDebt;
     }
     return { newDebt, toMint, fee };
   };
 
-  /** @param {ZCFSeat} clientSeat */
+  /**
+   * Adjust principal and collateral (atomically for offer safety)
+   *
+   * @param {ZCFSeat} clientSeat
+   */
```

Comment:
I don't see `updateDebtSnapshot()` in `adjustBalancesHook`. Shouldn't the debt be updated after we change the debt or collateral?